R/accessory.R

```diff
@@ -950,50 +950,63 @@ read_contrasts <-
 
     # Parse YAML contrasts into a data frame
     contrasts <- do.call(rbind, lapply(contrasts_yaml$contrasts, function(x) {
-      if (is.null(x$id)) {
-        stop("Missing contrast id in YAML contrasts.")
+    stopifnot(!is.null(x$id))  # Require that each contrast has an 'id'
+
+    row <- data.frame(
+      id = x$id,
+      variable = NA, reference = NA, target = NA,
+      blocking = NA,
+      formula = NA,
+      make_contrasts_str = NA,
+      stringsAsFactors = FALSE
+    )
+
+    if (!is.null(x$comparison)) {
```

Comment:
```suggestion
    )

    # A contrast can be defined in two ways:
    #
    # 1 . Specifying a column in the sample sheet, and two values from that column which define two groups of samples to compare. Tools using a contrast defined in this way would need to generate a model and contrast using that information.
    # 2. Specifying the formula and contrast string explicitly. Tools using those contrasts can then pass these directly to function of suites such as DESeq2. 

    if (!is.null(x$comparison)) {
```

Some sort of explanation of the two modes?